vcpkg/maintainers/functions/vcpkg_make_configure.md

```diff
@@ -0,0 +1,159 @@
+
+Prevents the addition of C Preprocessor flags to the compiler command line. Use this if the project does not use or incorrectly handles these flags.
+
+If set, the `CPPFLAGS` environment variable, typically used for specifying C preprocessor flags, will not be automatically populated based on vcpkg's settings. This can be useful for projects that do not use standard environment variables or have specific preprocessor requirements.
+
+### ADD_BIN_TO_PATH
+
+Adds the `bin` directory to the system path. This is useful for builds that require executables to be available on the path.
+
+### NO_DEFAULT_OPTIONS
+
+When set, this flag disables the automatic inclusion of the following default configure options provided by `vcpkg_make_configure`. Use this flag if you need full control over the configure command line:
+
+General Defaults:
+* --disable-silent-rules - Makes the build process output more verbose, which can be helpful for debugging.
+* --verbose - Enables verbose output from the configure script.
+
+Library Linkage:
+* For dynamic library builds: --enable-shared and --disable-static
+* For static library builds: --disable-shared and --enable-static
+
+Installation Directories:
+* Adjusts standard directories (bin, lib, share, etc.) to conform with vcpkg's layout, ensuring correct installation paths.
+
+Platform-Specific Tweaks:
+* On Windows: Ensures compatibility with Windows-specific filesystem and library behaviors.
+* On Unix-like systems: Assures the script accommodates the typical filesystem layout and toolchain behaviors.
+
+Removing these defaults gives you the responsibility to specify all necessary configuration options manually, ensuring the project is correctly configured for building with vcpkg.
+
+### NO_MSVC_FLAG_ESCAPING
+
+Turns off the escaping of MSVC flags. This might be necessary for projects that do not expect or handle escaped compiler flags properly.
+
+### USE_RESPONSE_FILES
+
+Enables the use of response files to pass arguments to the configure script. This can help avoid command line length limitations.
+
+### SOURCE_PATH
+
+The directory containing the project's source files
+
+This value is usually obtained as a result of calling a source acquisition command like [`vcpkg_from_github()`](vcpkg_from_github.md).
+
+### SHELL
+
+Specifies the shell to use for running the configuration scripts. Useful for ensuring compatibility in environments where the default shell might not behave as expected.
+
+### OPTIONS
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+### OPTIONS_RELEASE
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+These are in addition to `OPTIONS`.
+
+### OPTIONS_DEBUG
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+These are in addition to `OPTIONS`.
+
```

Comment:
Is it really necessary to document those one by one? Not even CMake is doing that.